File: leetcode/golang/LCR_179_he_wei_sde_liang_ge_shu_zi_lcof.go
func twoSum(price []int, target int) []int {
    i, j := 0, len(price) - 1
    for i < j {
        if price[i] + price[j] < target {
            i++
        } else if price[i] + price[j] > target {
            j--
        } else {
            return []int {price[i], price[j]}
        }
    }
    return []int{}
}

File: leetcode/c++/LCR_179_he_wei_sde_liang_ge_shu_zi_lcof.cpp
// https://leetcode.cn/problems/he-wei-sde-liang-ge-shu-zi-lcof/description

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& price, int target) {
        int i = 0, j = price.size() - 1;
        while (i < j) {
            if (price[i] + price[j] < target)
                i++;
            else if (price[i] + price[j] > target)
                j--;
            else
                return {price[i], price[j]};
        }
        return {};
    }
};

// O(n), O(1)